### ISSUE TITLE
git should provide a template commit message

### ISSUE THREAD

**Issue by PaulBone** [PlasmaLang/plasma#42]
Configure git to give devs a commit template message. We could also have a better example in CONTRIBUTING.md.

**Comment by Gertm** [comment]
I've been doing a bit of research for this.
There's a way to do this, but we need to write to the .git/config of this repository for that.
Maybe we could write a Makefile rule that writes the correct settings to the current git config? Could be a bit intrusive for the user, but on the other hand it ensures that our git template is used.

Do you want me to draw this up and make PR, so you can see for yourself?

**Comment by PaulBone** [comment]
I'm against doing something automatically to people's `.git/config`. I have had this done to me and it wasn't very plesent. I'd be happy to drop a template file into a `.git` directory and have that work the way it does with CVS (the last time I did something like this) but not modify a `.git/config` file.

When this happened to me it was a project that added a pre-commit script to lint the code. Which is good in theory, but it would end up committing things that I hadn't picked with "git add" and so it actually ended up committing code I hadn't intended to commit.  yarn or node or whatever tool it was that modified my config kept doing it.  I ended up doing `chmod 000 .git/config` to stop it.

So yeah, if this is the way to do it I'd rather provide some instructions for people to follow when they setup their workspace or a script they can run if they choose to, that tells them what it has just done to their configuration.

I don't know if you're missing something.  This is an area of git I havn't investigated yet.
